impl/client/src/main/javascript/web/pentaho/type/instance.js

```diff
@@ -253,7 +253,7 @@ define([
       _extend: function(name, instSpec, classSpec, keyArgs) {
         var typeSpec;
         if(name == null && (typeSpec = (instSpec && instSpec.type))) {
-          name = typeSpec.id || typeSpec.sourceId || null;
+          name = typeSpec.sourceId || typeSpec.id || null;
```

Comment:
The reason is that typically, the `sourceId` is a more faithful source for the name of a class. E.g. `pentaho/visual/base` vs `pentaho/visual/base/model`; the former would generate a class name of `pentaho.visual.Base` and the latter a class name of `pentaho.visual.base.Model`.
I'll update the comment.